src/timesignature.ts

```diff
@@ -106,26 +112,67 @@ export class TimeSignature extends StaveModifier {
     return new TimeSignatureGlyph(this, topDigits, botDigits, 'timeSig0', this.point);
   }
 
+  /**
+   * Returns line, num, glyph -- but these can be accessed directly w/ getters and setters.
+   */
   getInfo(): TimeSignatureInfo {
-    return this.info;
+    const { line, is_numeric, glyph } = this;
+    return { line, num: is_numeric, glyph };
   }
 
+  /**
+   * Set a new time signature specification without changing customPadding, etc.
+   *
+   * The getter for this is `getTimeSpec` not `getTimeSig`.
+   */
   setTimeSig(timeSpec: string): this {
-    this.info = this.parseTimeSpec(timeSpec);
+    this.timeSpec = timeSpec;
+    const info = this.parseTimeSpec(timeSpec);
+    this.glyph = info.glyph;
+    this.is_numeric = info.num;
+    this.line = info.line;
     return this;
   }
 
+  getTimeSpec(): string {
```

Comment:
Comments for the new functions are missing.

Reply:
will add -- I wasn't sure if we were doing them for getter/setter type methods, but will get.